server/protocols/webadmin/webadmin-data-jmap/src/main/java/org/apache/james/webadmin/data/jmap/MessageFastViewProjectionCorrector.java

```diff
@@ -128,15 +132,47 @@ boolean failed() {
     }
 
     private Mono<Void> correctMailboxProjectionItems(Progress progress, MessageManager messageManager, MailboxSession session) throws MailboxException {
-        return Iterators.toFlux(messageManager.getMessages(MessageRange.all(), FetchGroup.MINIMAL, session))
-            .concatMap(Throwing.function(messageResult -> Iterators.toFlux(messageManager.getMessages(MessageRange.all(), FetchGroup.BODY_CONTENT, session))))
-            .map(Throwing.function(messageResult -> Pair.of(messageResult.getMessageId(), projectionItemFactory.from(messageResult))))
-            .concatMap(pair -> Mono.from(messageFastViewProjection.store(pair.getKey(), pair.getValue()))
+        return listAllMailboxMessages(messageManager, session)
+            .concatMap(messageResult -> retrieveContent(messageManager, session, messageResult))
+            .map(this::computeProjectionEntry)
+            .concatMap(pair -> storeProjectionEntry(pair)
                 .doOnSuccess(any -> progress.processedMessageCount.incrementAndGet()))
             .onErrorContinue((error, triggeringValue) -> {
                 LOGGER.error("JMAP preview re-computation aborted for {} - {}", session.getUser(), triggeringValue, error);
                 progress.failedMessageCount.incrementAndGet();
             })
             .then();
     }
+
+    private Flux<MailboxMetaData> listUsersMailboxes(MailboxSession session) throws MailboxException {
+        return Flux.fromIterable(mailboxManager.search(MailboxQuery.privateMailboxesBuilder(session).build(), session));
+    }
+
+    private Mono<MessageManager> retrieveMailbox(MailboxSession session, MailboxMetaData mailboxMetadata) {
+        return Mono.fromCallable(() -> mailboxManager.getMailbox(mailboxMetadata.getId(), session));
+    }
+
+    private Flux<MessageResult> listAllMailboxMessages(MessageManager messageManager, MailboxSession session) throws MailboxException {
+        return Iterators.toFlux(messageManager.getMessages(MessageRange.all(), FetchGroup.MINIMAL, session));
+    }
+
+    private Flux<MessageResult> retrieveContent(MessageManager messageManager, MailboxSession session, MessageResult messageResult) {
+        try {
+            return Iterators.toFlux(messageManager.getMessages(MessageRange.one(messageResult.getUid()), FetchGroup.FULL_CONTENT, session));
+        } catch (MailboxException e) {
+            throw new RuntimeException(e);
```

Comment:
Is it what we really want or do we prefer `Flux.error(e)`  and handle it in `listAllMailboxMessages` ?